app/celery/letters_pdf_tasks.py

```diff
@@ -163,22 +167,83 @@ def letter_in_created_state(filename):
     return False
 
 
-@notify_celery.task(name='process-virus-scan-passed')
-def process_virus_scan_passed(filename):
+@notify_celery.task(bind=True, name='process-virus-scan-passed', max_retries=15, default_retry_delay=300)
+def process_virus_scan_passed(self, filename):
     reference = get_reference_from_filename(filename)
     notification = dao_get_notification_by_reference(reference)
     current_app.logger.info('notification id {} Virus scan passed: {}'.format(notification.id, filename))
 
     is_test_key = notification.key_type == KEY_TYPE_TEST
-    move_scanned_pdf_to_test_or_live_pdf_bucket(
+
+    scan_pdf_object = s3.get_s3_object(current_app.config['LETTERS_SCAN_BUCKET_NAME'], filename)
+    old_pdf = scan_pdf_object.get()['Body'].read()
+
+    new_pdf = _sanitise_precomiled_pdf(self, notification, old_pdf)
+
+    if not new_pdf:
+        current_app.logger.info('Invalid precompiled pdf received {} ({})'.format(notification.id, filename))
```

Comment:
I think it's worth switching 400s on template-preview so we start getting these log messages. The ones from template-preview itself don't seem to have anything that would help us find the original PDF.

Reply:
that's a v good point. I'll make a PR

Reply:
i have made a PR https://github.com/alphagov/notifications-template-preview/pull/193